<launch>
    <node name="hkcam_node" pkg="genicam" type="genicam_node" output="screen" >
        <param name="publish_topic" type="string" value="camera_front"/>
        <!-- 开启PTP同步之后可以使用device时间 -->
        <param name="enable_ptp" type="bool" value="True" />
        <param name="use_device_time" type="bool" value="True" />
        <!-- PTP时间戳为TAI时间，需转换为UTC时间，此处表示TAI时间比UTC提前多少 -->
        <param name="tai_ahead" type="int" value="37" />
        <param name="vis_img" type="bool" value="False"/>
        <param name="gev_packet_size" type="int" value="8164" />
        <param name="gev_packet_delay" type="int" value="20" />
        <param name="buffer_num" type="int" value="2" /> 
        <param name="identify_str" type="string" value="Hikrobot-MV-CS020-10GC-DA2690447"/>
        <param name="pixel_format" type="string" value="BayerRG8" />
        <param name="binning_h" type="int" value="2" />
        <param name="binning_w" type="int" value="2" />
        <param name="exposure_auto" type="string" value="Off" /> <!--Off/Once/Continuous-->
        <param name="exposure_time" type="double" value="4000" />
        <param name="gain_auto" type="string" value="Off" /> <!--Off/Once/Continuous-->
        <param name="gain" type="double" value="5" />
        <param name="triggermode" type="bool" value="True" />
    </node>
</launch>

